product/dialogs/miq_dialogs/miq_provision_google_dialogs_template.yaml

```diff
@@ -205,14 +205,18 @@
           :notes_display: :show
         :vm_name:
           :description: Instance Name
-          :required_method: :validate_vm_name
+          :required_method:
+            - :validate_vm_name
+            - :validate_regex
+          :required_regex: !ruby/regexp /\A(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?)\Z/
+          :required_regex_fail_details: The name must be composed only of lower-case English alphabet characters (a-z), numbers (0-9), and dashes (-). The first character must be a lower-case English alphabet character, and the last character must not be a dash.
```

Comment:
If you do the name validation after the name has been expanded it will processed on the task.  In that case the validation should really happen right before we send it to the provider so it can check any changes to the vm_name made from automate.  But the provider is already doing this validation for us at the task level and it seem odd to repeat it here.

I realized this morning there is a different approach we can take here.  The `:vm_name` field in the dialog has a `:notes` section which can be used to display text below the field.

Here is an example using the `:required_regex_fail_details` from this PR:
![image](https://cloud.githubusercontent.com/assets/2591569/17621784/b67e572c-6063-11e6-8d89-74d25d0a2071.png)

This could relieve the need for any of the validation checks.  Thoughts?